lib/spack/spack/cmd/install.py

```diff
@@ -227,10 +227,10 @@ def install_specs(cli_args, kwargs, specs):
                     tty.debug('{0} matched nothing in the env'.format(
                         abstract.name))
                     # no matches in the env
-                    if cli_args.no_add:
-                        msg = ('You asked to install {0} without adding it ' +
-                               '(--no-add), but no such spec exists in ' +
-                               'environment').format(abstract.name)
+                    if not cli_args.add:
+                        msg = ('You asked to install {0}, which is not in '
+                               'the current environment, but did not also '
+                               'specify --add'.format(abstract.name))
```

Comment:
I think this could be clearer. Maybe

"Cannot install '{0}' because it is not in the current environment. You can add it to the environment with 'spack add {0}', or as part of the install command with 'spack install --add {0}'.format(abstract)